File: python/reverse_string_words.py
# https://leetcode.com/problems/reverse-words-in-a-string

class Solution:
    def reverseWords(self, s):
        crumbs = s.split(' ')
        crumbs.reverse()
        new_list = []
        for item in crumbs:
            if item != '':
                new_list.append(item)

        return ' '.join(new_list)

File: javascript/reverse_string_words.js
// https://leetcode.com/problems/reverse-words-in-a-string

/**
 * @param {string} s
 * @return {string}
 */
var reverseWords = function(s) {
    crumbs = s.split(' ')
    crumbs.reverse()
    
    newList = []
    for (var i = crumbs.length - 1; i >= 0; i--) {
        if (crumbs[i] != '') {
            newList.push(crumbs[i]);
        }
    }
    
    return newList.reverse().join(' ');   
};

val = "the sky is blue"
val = "  hello world  "
val = "  Bob    Loves  Alice   "

out = reverseWords(val);
console.log(out);